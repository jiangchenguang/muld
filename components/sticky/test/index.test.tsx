import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup } from '@testing-library/react';
import Sticky from '..';

describe('Sticky', () => {
    afterEach(cleanup);

    it('sticky to top', () => {
        const { asFragment } = render(<Sticky>content</Sticky>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('z-index prop', () => {
        const { asFragment } = render(<Sticky zIndex={0}>content</Sticky>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('offset-top prop', () => {
        const { asFragment } = render(<Sticky offsetTop={10}>content</Sticky>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('offset-top with vw unit', () => {
        const { asFragment } = render(<Sticky offsetTop="10vw">content</Sticky>);
        expect(asFragment()).toMatchSnapshot();
    });
});
