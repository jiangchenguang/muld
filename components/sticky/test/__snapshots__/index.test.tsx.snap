// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticky offset-top prop 1`] = `
<DocumentFragment>
  .c0.mul-sticky .mul-sticky--fixed {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 99;
}

<div
    class="c0 mul-sticky"
    style="height: auto;"
  >
    <div
      class=""
    >
      content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sticky offset-top with vw unit 1`] = `
<DocumentFragment>
  .c0.mul-sticky .mul-sticky--fixed {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 99;
}

<div
    class="c0 mul-sticky"
    style="height: auto;"
  >
    <div
      class=""
    >
      content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sticky sticky to top 1`] = `
<DocumentFragment>
  .c0.mul-sticky .mul-sticky--fixed {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 99;
}

<div
    class="c0 mul-sticky"
    style="height: auto;"
  >
    <div
      class=""
    >
      content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sticky z-index prop 1`] = `
<DocumentFragment>
  .c0.mul-sticky .mul-sticky--fixed {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 99;
}

<div
    class="c0 mul-sticky"
    style="height: auto;"
  >
    <div
      class=""
    >
      content
    </div>
  </div>
</DocumentFragment>
`;
